### DISCUSSION TITLE
How to integrate Stellar with WalletConnect

### DISCUSSION THREAD

**Discussion by elionntechguy** [WalletConnect/walletconnect-monorepo#767] [Developer Support] [answered]
Hello, any ideas on how to integrate stellar with WalletConnect? I've tried everything, can't seem to get it to work, i'm trying to connect to LOBSTR. This is the session permissions set:
`permissions: {
        blockchain: {
          chains: ["stellar:pubnet"],
        },
        jsonrpc: {
          methods: ["stellar_signAndSubmitXDR", "stellar_signXDR"],
        },
      },`

Edit: 
Found a solution, view @MarcoCiaramella's reply below.

**Comment by MarcoCiaramella** [comment, marked as answer]
stellar chain is supported by v2.0beta https://docs.walletconnect.com/2.0/quick-start/dapps/client. Refer to https://github.com/stellarterm/stellarterm project. It uses walletconnect for pairing the lobstr wallet

**Reply by elionntechguy** [reply]
I've explored the stellarterm repo, i've gotten as far as the qrcode modal opening, but when trying to scan the QRCode, it doesn't connect. Nothing pops up.

**Reply by chadyj** [reply]
@elionntechguy Take a look at https://stellarterm.zendesk.com/hc/en-us/articles/4406496403217-WalletConnect-How-do-I-log-in-to-StellarTerm-with-my-wallet-app- if you are having issues scanning the QR code.

**Reply by elionntechguy** [reply]
> @elionntechguy Take a look at https://stellarterm.zendesk.com/hc/en-us/articles/4406496403217-WalletConnect-How-do-I-log-in-to-StellarTerm-with-my-wallet-app- if you are having issues scanning the QR code.


I think the problem i'm having is with my code, theoretically the code should be working, there are no errors, but when i get to the QRCode modal, and try to scan with LOBSTR, it doesn't get to the session proposal, it just doesn't connect, the LOBSTR QRCode scanner for walletconnect closes.

I can send you the code if you want.

**Reply by elionntechguy** [reply]
@chadyj 

This basically describes the problem.

BTW i'm using WalletConnect 2.0 Beta

https://user-images.githubusercontent.com/41442942/153912794-81753a2f-1679-4ae0-a699-c96e01ed3c2b.mp4

**Reply by MarcoCiaramella** [reply]
see my [gist](https://gist.github.com/MarcoCiaramella/097b28e308817b2b7143d95a5e8d84f2)

**Reply by elionntechguy** [reply]
@MarcoCiaramella This works! Thank you so much!

**Reply by MarcoCiaramella** [reply]
@elionntechguy you are welcome :)

**Reply by FosterSOAsare** [reply]
@MarcoCiaramella  there is an error in your gist when I try it. It says client doesn't provide an export named CLIENT_EVENTS is there a way around it?

**Comment by alejomendoza** [comment]
How do you integrate it with V2?

we can't connect now using the following code:

```
const { uri, approval } = await client.connect({
  requiredNamespaces: {
    stellar: {
      methods: ['stellar_signAndSubmitXDR', 'stellar_signXDR'],
      chains: ['stellar:pubnet'],
      events: ['chainChanged', 'accountsChanged'],
    },
  },
})
```

**Reply by BenRacicot** [reply]
I too would love an example, perhaps with Stellar's SDK to create a simple payment transaction and sign it with the permissioned WalletConnected wallet.

**Reply by hackyguru** [reply]
We don't have a code snippets specific to Stellar yet but we'll definitely take your feedback and work on adding them. Till then, can you check out the RPC docs here - https://docs.walletconnect.com/2.0/advanced/rpc-reference/stellar-rpc
